graspologic/inference/group_connection_test.py

```diff
@@ -0,0 +1,325 @@
+                Same as above, but for network 2.
+            "observed1" = n_observed1, dataframe
+                The total number of observed group-to-group edge connections for network 1.
+            "observed2" = n_observed2, dataframe
+                Same as above, but for network 2.
+            "possible1" = n_possible1, dataframe
+                The total number of possible edges for each group-to-group pair in network 1.
+            "possible2" = n_possible2, dataframe
+                Same as above, but for network 2.
+            "group_counts1" = group_counts1, pd.series
+                Contains total number of nodes corresponding to each group label for network 1.
+            "group_counts2" = group_counts2, pd.series
+                Same as above, for network 2
+            "null_ratio" = adjustment_factor, float
+                If the "density adjustment" parameter is set to "true", this variable contains the null hypothesis for the quotient of
+                odds ratios for the group-to-group connection densities for the two networks. In other words, it contains the hypothesized
+                factor by which network 1 is "more dense" or "less dense" than network 2. If "density adjustment" is set to "false", this
+                simply returns a value of 1.0.
+            "n_tests" = n_tests, integer
+                This variable contains the number of group-to-group comparisons performed by the function.
+            "rejections" = rejections, dataframe
+                Contains a square matrix of boolean variables. The side length of the matrix is equal to the number of distinct group
+                labels. An entry in the matrix is "true" if the null hypothesis, i.e. that the group-to-group connection density
+                corresponding to the row and column of the matrix is equal for both networks (with or without a density adjustment factor),
+                is rejected. In simpler terms, an entry is only "true" if the group-to-group density is statistically different between
+                the two networks for the connection from the group corresponding to the row of the matrix to the group corresponding to the
+                column of the matrix.
+            "corrected_pvalues" = corrected_pvalues, dataframe
+                Contains the p-values for the group-to-group connection densities after correction using the chosen correction_method.
+
+
+    Notes
+    ------
+    The function name has been changed to group_connection_test to match the chosen language in the paper.
+    """
+
+    B1, n_observed1, n_possible1, group_counts1 = fit_sbm(A1, labels1)
+    B2, n_observed2, n_possible2, group_counts2 = fit_sbm(A2, labels2)
+
+    # TODO fix this up
+    assert n_observed1.index.equals(n_observed2.index)
+    assert n_observed1.columns.equals(n_observed2.columns)
+    assert n_possible1.index.equals(n_possible2.index)
+    assert n_observed1.columns.equals(n_possible2.columns)
```

Comment:
check that the community labels have the same communities, raise value error if not. this can happen before fitting the sbms